SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialLoadBalanceEnforcer.cpp

```diff
@@ -75,26 +103,24 @@ void USpatialLoadBalanceEnforcer::ProcessQueuedAclAssignmentRequests()
 
 	for (WriteAuthAssignmentRequest& Request : AclWriteAuthAssignmentRequests)
 	{
-		static const int32 WarnOnAttemptNum = 5;
-		if (Request.ProcessAttempts >= WarnOnAttemptNum)
-		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Failed to process WriteAuthAssignmentRequest with EntityID: %lld. Process attempts made: %d"), Request.EntityId, Request.ProcessAttempts);
-		}
-
-		Request.ProcessAttempts++;
-
-		// TODO - if some entities won't have the component we should detect that before queueing the request.
-		// Need to be certain it is invalid to get here before receiving the AuthIntentComponent for an entity, then we can check() on it.
-		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
+ 		const AuthorityIntent* AuthorityIntentComponent = StaticComponentView->GetComponentData<SpatialGDK::AuthorityIntent>(Request.EntityId);
 		if (AuthorityIntentComponent == nullptr)
 		{
-			UE_LOG(LogSpatialLoadBalanceEnforcer, Warning, TEXT("Detected entity without AuthIntent component. EntityId: %lld"), Request.EntityId);
-			continue;
+			// TODO(zoning): This is possible if the ACL update lags behind the intent update. Evaluate whether this is a serious problem.
```

Comment:
Should this be the other way round? i.e. if the intent lags behind the ACL? This check failing implies in intent doesn't exist locally?

Reply:
Also I would've thought this was impossible since the AuthorityIntent should always be received in the initial critical section?

Reply:
The code here actually gets hit on worker A when authority is transferring from A to B. A has shipped off the intent authority, but it still has the ACL authority.

Reply:
But in that case doesn't it still have interest in the entity. In which case the authority intent component can't be null.

Reply:
hmm, I'm not sure. I do know that I was hitting this case, but maybe another fix since then has made them not happen. I'll test this morning.

Reply:
Would think the same as what @samiwh suggests.

Reply:
So I'm not seeing the warning now when I run tests, but I know I did see it previously. I'm inclined to leave the warning in the code for now, and update the comment to "Remove this if we're not seeing the warning"